client/src/components/QandA/QuestionList.jsx

```diff
@@ -2,31 +2,77 @@
 import React from 'react';
 import QuestionListEntry from './QuestionListEntry.jsx';
 import AddQuestion from './AddQuestion.jsx';
+import { answerFormatter } from './Utilities.js';
 
-const QuestionList = (props) => {
-  return (
-    <div className="container">
-      <p className="text-start fs-5 fw-light">
-      Questions and Answers
-      </p>
-      <div className="input-group mb-3">
-        <input type="text" className="form-control" placeholder="HAVE A QUESTION? SEARCH FOR ANSWERS..." aria-label="Recipient's username" aria-describedby="button-addon2"></input>
-        <button className="btn btn-outline-secondary" type="button" id="button-addon2">Filter Questions</button>
-        {/* Above button label will be replaced by a magnifying glass icon in later update FIX ME ^^ */}
-      </div>
-      <div className="row">
-        {props.data.results.map((question, index) => {
-          if (index < 2) {
-            return <QuestionListEntry key={index} data={question} currentProductName={props.currentProductName} answers={props.answerParser(question.answers)}/>;
-          }
-          if (index >= 2) {
-            // return <LoadMoreQuestions data={question} answers={props.answerParser(question.answers)} />; <-- add LoadMoreQuestions component
+class QuestionList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      questionsShown: 2,
+      searchBarInput: '',
+      questionList: this.props.data
```

Comment:
You might want to change this.props.data to something more descriptive later.

Reply:
Duly noted!